src/PVRIptvData.cpp

```diff
@@ -468,10 +475,11 @@ bool PVRIptvData::LoadPlayList(void)
 
       iChannelNum++;
 
-      if (iCurrentGroupId > 0)
+      std::vector<int>::iterator it;
+      for (it = iCurrentGroupId.begin(); it != iCurrentGroupId.end(); ++it)
       {
-        channel.bRadio = m_groups.at(iCurrentGroupId - 1).bRadio;
-        m_groups.at(iCurrentGroupId - 1).members.push_back(iChannelIndex);
+        channel.bRadio = m_groups.at(*it - 1).bRadio;
```

Comment:
I'm not sure that `*it - 1` will not point to nirvana if there is only one group.